server/persistence/dynamo/dynamo_service_interaction.go

```diff
@@ -0,0 +1,244 @@
+package dynamo
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+)
+
+const (
+	interactionInfoAttributeName   = `info`
+	interactionPreferAttributeName = `prefer`
+)
+
+var _ persistence.InteractionService = (*interactionService)(nil)
+
+type interactionService struct {
+	ctx context.Context
+
+	svc *dynamodb.Client
+}
+
+func newInteractionService(
+	ctx context.Context,
+	svc *dynamodb.Client,
+) persistence.InteractionService {
+	return &interactionService{
+		ctx: ctx,
+		svc: svc,
+	}
+}
+
+func (is *interactionService) Get(
+	id model.PlayerID,
+) (interaction.PlayerMeans, error) {
+	pkName := `:ipID`
+	pk := string(id)
+	skName := `:sk`
+	sk := getSortKeyPrefix(is)
+	keyCondExpr := getConditionExpression(equalsID, pkName, hasPrefix, skName)
+
+	createQuery := getPkSkCondCreateQuery(pk, pkName, sk, skName, keyCondExpr)
+	items, err := fullQuery(is.ctx, is.svc, createQuery)
+	if err != nil {
+		return interaction.PlayerMeans{}, err
+	}
+
+	return is.buildPlayerMeansFromItems(id, items)
+}
+
+func (is *interactionService) buildPlayerMeansFromItems(
+	id model.PlayerID,
+	items []map[string]types.AttributeValue,
+) (interaction.PlayerMeans, error) {
+
+	pm := interaction.PlayerMeans{
+		PlayerID:      id,
+		PreferredMode: interaction.Unknown,
+	}
+
+	for _, item := range items {
+		if preferAV, ok := item[is.getPreferKey()]; ok {
+			if pm.PreferredMode != interaction.Unknown {
+				return interaction.PlayerMeans{}, errors.New(`preferred mode already set`)
+			}
+
+			preferAVN, ok := preferAV.(*types.AttributeValueMemberN)
+			if !ok {
+				return interaction.PlayerMeans{}, errors.New(`wrong prefer attribute type`)
+			}
+			preferMode, err := strconv.Atoi(preferAVN.Value)
+			if err != nil {
+				return interaction.PlayerMeans{}, err
+			}
+
+			pm.PreferredMode = interaction.Mode(preferMode)
+			continue
+		}
+
+		mode, serInfo, err := is.getInteractionModeAndSerInfo(
+			item[sortKey],
+			item[is.getInfoKey()],
+		)
+		if err != nil {
+			return interaction.PlayerMeans{}, err
+		}
+
+		m := interaction.Means{
+			Mode: mode,
+		}
+		err = m.AddSerializedInfo(serInfo)
+		if err != nil {
+			return interaction.PlayerMeans{}, err
+		}
+
+		pm.Interactions = append(pm.Interactions, m)
+
+	}
+
+	return pm, nil
+}
+
+func (is *interactionService) getInteractionModeAndSerInfo(
```

Comment:
This function does two independent things. Should we be trying to be more SOLID here?